### PR TITLE
feat:Filter permissions based on case access category(RWA-924)

### PR DESCRIPTION
**Pull request by dharmendrak**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-924

### Change description ###

Filter permissions based on case access category

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by serdaroguzhanakin — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/CaseConfigurationProviderServiceTest.java [hunk 1]
```diff
@@ -156,4 +160,232 @@ void gets_fields_to_map() {
...
+
+
+        TaskConfigurationResults mappedData = caseConfigurationProviderService
+            .getCaseRelatedConfiguration(someCaseId, Map.of());
+
+        assertThat(mappedData.getPermissionsDmnResponse(), is(permissions));
+    }
+
+    @Test
+    void should_consider_permissions_when_case_access_category_column_does_not_matches() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/CaseConfigurationProviderServiceTest.java:279]: can we rename this to match the behaviour and outcome? e.g should_return_empty_when.........

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/CaseConfigurationProviderServiceTest.java [hunk 2]
```diff
@@ -156,4 +160,232 @@ void gets_fields_to_map() {
...
+
+
+        TaskConfigurationResults mappedData = caseConfigurationProviderService
+            .getCaseRelatedConfiguration(someCaseId, Map.of());
+
+        Assertions.assertThat(mappedData.getPermissionsDmnResponse()).isEmpty();
+    }
+
+    @Test
+    void should_consider_permissions_when_case_access_category_column_is_present_but_case_field_is_null() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/CaseConfigurationProviderServiceTest.java:317]: can we rename this to match the behaviour and outcome? e.g should_return_empty_when.........

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/CaseConfigurationProviderServiceTest.java [hunk 3]
```diff
@@ -156,4 +160,232 @@ void gets_fields_to_map() {
...
+
+
+        TaskConfigurationResults mappedData = caseConfigurationProviderService
+            .getCaseRelatedConfiguration(someCaseId, Map.of());
+
+        Assertions.assertThat(mappedData.getPermissionsDmnResponse()).isEmpty();
+    }
+
+    @Test
+    void should_consider_permissions_when_case_access_category_column_is_present_but_case_field_is_empty() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/CaseConfigurationProviderServiceTest.java:353]: can we rename this to match the behaviour and outcome? e.g should_return_empty_when.........

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskInitiateByIdControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+        assertions.taskVariableWasUpdated(
+            taskVariables.getProcessInstanceId(),
+            "cftTaskState",
+            "unassigned"
+        );
+
+        common.cleanUpTask(taskId);
+    }
+
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskInitiateByIdControllerTest.java:162]: we have no test to show the caseCategory AC's?  I.e. would be for a WA caseType

**Review by serdaroguzhanakin — APPROVED** [review]

**Pull request merged**